File: ios/BasicQRScanner/MLKit/MLKitViewManager.m
//
//  VisionKitViewManager.m
//  BasicQRApp
//
//  Created by Ananth Desai on 31/12/24.
//

#import <Foundation/Foundation.h>
#import "React/RCTViewManager.h"

@interface RCT_EXTERN_MODULE(MLKitViewManager, RCTViewManager)
  RCT_EXPORT_VIEW_PROPERTY(onSuccessfulScan, RCTDirectEventBlock)
@end

File: ios/BasicQRScanner/MLKit/MLKitViewManager.swift
//
//  MLKitViewManger.swift
//  BasicQRApp
//
//  Created by Ananth Desai on 31/12/24.
//

import Foundation

@objc(MLKitViewManager)
class MLKitViewManager: RCTViewManager {
  private weak var scannerRef: MLKitView? = nil
  override class func requiresMainQueueSetup() -> Bool {
    false
  }
  
  override func view() -> UIView! {
    let ref = MLKitView()
    scannerRef = ref
    return ref
  }
}
